lib/compat/wordpress-6.1/block-editor-settings.php

```diff
@@ -13,6 +13,23 @@
  * @return array New block editor settings.
  */
 function gutenberg_get_block_editor_settings( $settings ) {
+	global $post_id;
+	$template_slug    = get_page_template_slug( $post_id );
+	$current_template = gutenberg_get_block_templates( array( 'slug__in' => array( $template_slug ) ) );
+
+	if ( ! empty( $current_template ) ) {
+		$template_blocks    = parse_blocks( $current_template[0]->content );
+		$post_content_block = array();
+
+		foreach ( $template_blocks as $template_block ) {
+			if ( 'core/post-content' === $template_block['blockName'] ) {
+				$post_content_block = $template_block;
+			}
+		}
+		// mismatched naming x(.
+		$post_content_block['attributes'] = $post_content_block['attrs'];
+		$settings['postContentBlock']     = $post_content_block;
```

Comment:
Instead of adding the post content block to the settings, can we add the right layout directly instead and avoid all JS computations?

Reply:
It's complicated 😅 

In the visual editor component we need to add both the relevant layout classes and any styles resulting from user customisation (the ones we usually attach to `wp-container` classes in blocks). Currently, classes and styles are computed by `useLayoutClasses` and `useLayoutStyles` hooks, both of which take in the full block because at some point they need to know the block name.

Another consideration is that eventually we also want other Post Content settings such as typography to be reflected in the editor, so if we already have access to the whole block here it'll allow us to add that easily.

Reply:
It doesn't make sense to tie `useLayoutClasses` and `useLayoutStyles` to blocks. It seems we may be creating dependencies between different concepts that are just breaking the abstractions.

Why do we need blocks in these two functions?

Reply:
I guess maybe we need the "config" of the layout block support, in that case, it should probably be a separate argument no?

Reply:
For `useLayoutClasses` we use the block name to work out what the default layout for that block is. We need this because the layout attribute will be empty unless the block has non-defaults set.

For `useLayoutStyles` we also use the block name to pass to `getLayoutStyle`, which uses it to check for block gap support. 

I don't think we're creating dependencies here as much as working with existing dependencies 😅 but I'm not sure that's such a bad thing in this case - would we want to leverage those functions for non-block use cases? Does it make sense to abstract them from the block scenario unless we already have other potential use-cases to work with?

Reply:
> Does it make sense to abstract them from the block scenario unless we already have other potential use-cases to work with?

Isn't the root layout a use-case for a non block case? I mean the post-content block is not even part of the edited entity. 

Also, it seems the dependency is actually not the block object (like being passed in this PR) but the block name, so would it be better to pass only the block name (from the server there's no need for block name, I guess it's always core/post-content). Alternatively, we can also pass the "layout block support" config and not the block name.

Reply:
> Isn't the root layout a use-case for a non block case? I mean the post-content block is not even part of the edited entity.

It's not really a non-block case, because the root layout in the post editor is essentially the content of the Post Content block 😅 What we need to access here is the settings of the specific Post Content block that will render the post being edited, so we're still working with a block.

> Also, it seems the dependency is actually not the block object (like being passed in this PR) but the block name, so would it be better to pass only the block name (from the server there's no need for block name, I guess it's always core/post-content). Alternatively, we can also pass the "layout block support" config and not the block name.

I guess for those functions we could pass the layout object and the block name. We do need the block name though, because in the layout type's `getLayoutStyle` we need to check if the block skips serialisation for gap values.

Even then, if we just add the layout object to the editor settings, later we'll have to add the typography styles object too, so any custom Post Content typography styles are reflected in the editor. And if we ever add other block supports to Post Content, we'll have to do the same for each individually. It seems easier to just add the whole block so we can grab whatever styles and settings we need from it later.